### PR TITLE
rework external_url

### PR DESCRIPTION
**Pull request by cristian-radu**
- **What does the PR do?** *(JIRA reference, background context, references to other tickets)*
- **What automated testing exists for this change?** *Link to new or existing unit tests*
- **Is there documentation?** *If yes, please provide a link to the docs.prsn.io page or appropriate Readme*
- **Any dependencies?** *(PRs in other repos that need to be merged alongside this)*
- [x] **Has the changelog been updated?**

### CODE REVIEW

File: pkg/reaper/reaper.go [hunk 1]
```diff
@@ -30,10 +30,21 @@ func (r *Reaper) Cleanup(cfg *bitesize.Environment) error {
...
+		// orphaned services
 		if cfg.Services != nil && cfg.Services.FindByName(service.Name) == nil {
 			log.Infof("REAPER: Found orphan service %s, deleting.", service.Name)
 			r.deleteService(service)
 		}
+		// delete ingresses that were removed from the service config
+		// service is defined in config
+		if cfg.Services.FindByName(service.Name) != nil &&
+			// doesn't have an external_url in config
+			len(cfg.Services.FindByName(service.Name).ExternalURL) == 0 &&
```
Comment (geriBatai) [inline pkg/reaper/reaper.go]: Good overall. I'd just do it like this:
```
 kubesvc := cfg.Services.FindByName(service.Name)

if kubesvc != nil && kubesvc.HasExternalUrl() && !service.HasExternalURL() {
...
}
```
And move it to `CleanupIngress(service)`. Just makes it more readable and extendable.

Comment (geriBatai) [reply pkg/reaper/reaper.go]: Also, don't see test for that but I suppose it's in previous commits?

File: pkg/cluster/cluster.go [hunk 2]
```diff
@@ -130,11 +130,9 @@ func (cluster *Cluster) ApplyEnvironment(currentEnvironment, newEnvironment *bit
 			}
 
 			if len(service.ExternalURL) != 0 {
```
Comment (geriBatai) [inline pkg/cluster/cluster.go]: we can use HasExternalURL() now :)

Comment (cristian-radu) [reply pkg/cluster/cluster.go]: yeah, OK. I missed that one :)

**Comment by geriBatai** [comment]
LGTM

**Pull request merged**